Mod/Villains/Leviathan/Cards/TorrentialDownpourCardController.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Handelabra.Sentinels.Engine.Controller;
+using Handelabra.Sentinels.Engine.Model;
+using System.Collections;
+
+namespace Jp.ParahumansOfTheWormverse.Leviathan
+{
+
+    // Whenever a player would draw a card, Leviathan deals that player's character card 1 cold damage. This card is indestructible
+
+    public class TorrentialDownpourCardController : CardController
+    {
+        public TorrentialDownpourCardController(Card card, TurnTakerController controller) : base(card, controller)
+        {
+            //This card is indestructible
+            base.AddThisCardControllerToList(CardControllerListType.MakesIndestructible);
+        }
+
+        public override void AddTriggers()
+        {
+            //Whenever a player would draw a card...  
+            AddTrigger<DrawCardAction>(
+                dca => dca.HeroTurnTaker != null,
+                dca => RespondToCardDraw(dca),
+                TriggerType.DealDamage,
+                TriggerTiming.Before
+
+            /*
+             Trigger is triggering off drawing card, but IS dealing damage - unclear whether to use TriggerType.DealDamage or TriggerType.DrawCard

```

Comment:
It's for the effect of the trigger, so yeah it's DealDamage. The engine uses it to order triggers + determine when to ask the players to order triggers.